narwhals/dataframe.py

```diff
@@ -745,16 +763,52 @@ def schema(self) -> dict[str, DType]:
             You can pass either pandas or Polars to `func`:
 
             >>> df_pd_schema = func(df_pd)
-            >>> df_pd_schema
-            {'foo': Int64, 'bar': Float64, 'ham': String}
+            >>> df_pd_schema  # doctest:+SKIP
+            Schema({'foo': Int64, 'bar': Float64, 'ham', String})
```

Comment:
it could be argued that this is a breaking change...although `Schema` is a subclass of `OrderedDict`, and seems to be behaved the same way

I don't think there's any code written currently which would break under this change? If so, then I think this fits in with the stable API spirit, and we can do it

Reply:
> it could be argued that this is a breaking change...although Schema is a subclass of OrderedDict, and seems to be behaved the same way

`Schema` adds methods to an `OrderedDict`, I think we are fairly fine

> I don't think there's any code written currently which would break under this change? If so, then I think this fits in with the stable API spirit, and we can do it

So... I was wondering how the stable api would work. We can _only_ add, but never change or delete?!

Reply:
we need to make sure that any code written with `v1` doesn't break - so, we can remove from `narwhals`, but keep it around in `v1`

I think this case is fine anyway, as ordereddict is still a dict 😎